feat(color-picker): add color-picker component

Hades-li opened the pull request:
add color-picker component

<!-- Specify your pull request type -->
- [ ] Component Migration

<!-- Specify the component migration issue -->
Fix [https://github.com/element-plus/element-plus/issues/132](https://github.com/element-plus/element-plus/issues/132)

dependabot-preview[bot] committed:
fix: finish unit test, review code, fixed some bug

Bumps [@storybook/html](https://github.com/storybookjs/storybook/tree/HEAD/app/html) from 6.0.20 to 6.0.21.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.0.21/app/html)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 3.10.1 to 4.0.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.0.0/packages/eslint-plugin)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump husky from 4.2.5 to 4.3.0

Bumps [husky](https://github.com/typicode/husky) from 4.2.5 to 4.3.0.
- [Release notes](https://github.com/typicode/husky/releases)
- [Changelog](https://github.com/typicode/husky/blob/master/CHANGELOG.md)
- [Commits](https://github.com/typicode/husky/compare/v4.2.5...v4.3.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump html-webpack-plugin from 4.3.0 to 4.4.1

Bumps [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin) from 4.3.0 to 4.4.1.
- [Release notes](https://github.com/jantimon/html-webpack-plugin/releases)
- [Changelog](https://github.com/jantimon/html-webpack-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jantimon/html-webpack-plugin/compare/v4.3.0...v4.4.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

ci: fix dependabot automerge

build(deps-dev): bump @vue/compiler-sfc from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps): bump vue from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps-dev): bump @swc-node/jest from 0.2.0 to 0.3.4

build(deps-dev): bump @storybook/addon-storysource from 6.0.20 to 6.0.21

build(deps-dev): bump @types/jest from 26.0.10 to 26.0.13

build(deps-dev): bump highlight.js from 10.1.2 to 10.2.0

build(deps-dev): bump sass-loader from 10.0.1 to 10.0.2

build(deps-dev): bump vue-router from 4.0.0-beta.7 to 4.0.0-beta.9

build(deps-dev): bump file-loader from 6.0.0 to 6.1.0

build(deps): bump dayjs from 1.8.34 to 1.8.35

build(deps-dev): bump lint-staged from 10.2.13 to 10.3.0

build(deps-dev): bump eslint from 7.7.0 to 7.8.1

build(deps-dev): bump @babel/core from 7.11.4 to 7.11.6

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.11.4 to 7.11.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.11.6/packages/babel-core)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

refactor(transfer): transfer component refactoring completed

docs(transfer): add docs files

test(transfer): add test file

refactor(transfer): correct requested changes

fix(transfer): rename d.ts to .ts

fix(transfer): remove unused declare

fix(transfer): update test cases

fix(steps): fix error in deploy

refactor(steps): re-add change event

feat(utils): add change event constant to constants.ts

refactor(steps): correct requested changes

refactor(steps): replace indexOf with includes

build(deps-dev): bump vue-loader from 16.0.0-beta.5 to 16.0.0-beta.6

build(deps-dev): bump css-loader from 4.2.2 to 4.3.0

build(deps-dev): bump @swc-node/jest from 0.3.4 to 0.3.5

build(deps-dev): bump vue-loader from 16.0.0-beta.6 to 16.0.0-beta.7

test: hoist up mocks (#294)

### feat: add dialog (#197)

* Add overlay component; Dialog component almost done

* feat(dialog): add use-lockscreen

* feat(dialog): coding completed awaiting tests

* feat(dialog): finish writing test cases

* fix test failures

* Address PR comments

* fallback some changes

fix(dialog): fix dialog exporting error (#298)

fix: fix lang type error (#299)

fix: fix eslint && typescript-eslint issue

fix(input): fix issues

https://github.com/ElemeFE/element/issues/18573

fix:Fit Avatar Container style

feat: inifinite scroll done

fix(infinite-scroll): wipe out sleep

fix: merge error

fix(infinite-scroll): should not trigger load when scroll up

Feat/carousel (#151)

* feat: init carousel component

* feat(carousel): init carousel component

* chore: add global dependencies

* chore: use prettier formate code

* feat(carousel): transfer logic - progress 40%

* feat(carousel): migrate logic - progress 50%

* feat(carousel): migrate logic - progress 55%

* feat(carousel): replenish doc

* feat(carousel): add utils

* feat(carousel): finish component logi

* feat(carousel): fix type error

* feat(carousel): test case 80%

* feat(carousel): migrate finish

* feat(carousel): more test cases

* feat(carousel): test case passed

* feat(carousel): fix CI dependencies collides

* feat(carousel): update yarn.lock

* feat(carousel): merge sub component

* feat(carousel): fix lose ctx attribute in buid env

* feat(carousel): finish spec

* feat(carousel): optimize code

* chore: update yarn lock

* feat(carousel): fall back lock file

* chore(carousel): fallback dep

* feat(carousel): update vue dep

* feat(carousel): update spec file

* feat(carousel): use async test

* feat(carousel): revert eslint modify

Co-authored-by: liik <385211478@qq.com>
Co-authored-by: liik <linyunqianpp@126.com>

fix(tag): fix click event of tag

feat(core): add jsx support

build: init build script

chore: update

1.0.1-alpha.0

chore: update

chore: update

chore: update

chore: update

1.0.1-alpha.1

chore: update

style(dropdown): improve dropdown style

partial fix

fix(popper): fix known bugs in popper

fix(dropdown): fix dropdown code based on popper's change

fix(popper): addressing comments

fix: Update install doc

chore: update

fix: Fix website code highlight

feat: Feature/timepicker && repeat-click directive (#289)

* feat: Feature/datepicker && repeat-click directive (#288)

* style: fix lint

* test: fix local test

* test: update test

* fix: update api to disabledHours

* chore: update

* chore: fix lint

feat: add calendar (#305)

* feat: add calendar

* chore: update doc

* chore: update

* chore: update

fix(locale): i18n support params

Fix/update website (#322)

* 1.0.1-alpha.2

* chore: update website logo

* chore: update website logo

chore: update readme (#325)

Feature/drawer (#248)

* feat(drawer): feature is done without test

* test(drawer): add unit test for drawer and fix drawer style

* fix(drawer): lint error for revert eslintrc

* feat(drawer): optimize appendtobody by use built-in teleport

* fix(drawer): resolve some review conservation

* fix(drawer): resolve nest drawer & use v-model for default visible prop

Co-authored-by: luxueyan <luxueyan@kaitongamc.com>

feat: bump vue version (#330)

* feat(core): bump vue version

* fix(steps): remove the usage of setupState

feat: init type

feat(hooks): add useAttrs hook (#317)

* feat(hooks): add useAttrs hook

* feat(hooks): undo binding class and style automatically

* test: remove unused import statement

fix: carousel style (#328)

fix(popper): fix popper styling bug/adding transition events (#334)

* ammend

* fix judgement

refactor(project): replace keyCode (#337)

fix: change the value of version from 'indenpendent' to 'independent' (#338)

fix(core): remove unused files (#341)

chore(core): deprecate storybook (#343)

feat(color-picker): add color-picker component

add color-picker component

perf: delete test.vue file, add type defintion

add 'IUseOptions' interface for provide and inject function.  del test.vue component

fix: add modelValue watch. fixed some problem for review code

fix: finish unit test, review code, fixed some bug

fix: fixed unit test bug and components bug

fix(color-picker): add unit test and rebase from dev

Bumps [@storybook/html](https://github.com/storybookjs/storybook/tree/HEAD/app/html) from 6.0.20 to 6.0.21.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.0.21/app/html)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 3.10.1 to 4.0.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.0.0/packages/eslint-plugin)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump husky from 4.2.5 to 4.3.0

Bumps [husky](https://github.com/typicode/husky) from 4.2.5 to 4.3.0.
- [Release notes](https://github.com/typicode/husky/releases)
- [Changelog](https://github.com/typicode/husky/blob/master/CHANGELOG.md)
- [Commits](https://github.com/typicode/husky/compare/v4.2.5...v4.3.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump html-webpack-plugin from 4.3.0 to 4.4.1

Bumps [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin) from 4.3.0 to 4.4.1.
- [Release notes](https://github.com/jantimon/html-webpack-plugin/releases)
- [Changelog](https://github.com/jantimon/html-webpack-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jantimon/html-webpack-plugin/compare/v4.3.0...v4.4.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

ci: fix dependabot automerge

build(deps-dev): bump @vue/compiler-sfc from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps): bump vue from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps-dev): bump @swc-node/jest from 0.2.0 to 0.3.4

build(deps-dev): bump @storybook/addon-storysource from 6.0.20 to 6.0.21

build(deps-dev): bump @types/jest from 26.0.10 to 26.0.13

build(deps-dev): bump highlight.js from 10.1.2 to 10.2.0

build(deps-dev): bump sass-loader from 10.0.1 to 10.0.2

build(deps-dev): bump vue-router from 4.0.0-beta.7 to 4.0.0-beta.9

build(deps-dev): bump file-loader from 6.0.0 to 6.1.0

build(deps): bump dayjs from 1.8.34 to 1.8.35

build(deps-dev): bump lint-staged from 10.2.13 to 10.3.0

build(deps-dev): bump eslint from 7.7.0 to 7.8.1

build(deps-dev): bump @babel/core from 7.11.4 to 7.11.6

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.11.4 to 7.11.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.11.6/packages/babel-core)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

refactor(transfer): transfer component refactoring completed

docs(transfer): add docs files

test(transfer): add test file

refactor(transfer): correct requested changes

fix(transfer): rename d.ts to .ts

fix(transfer): remove unused declare

fix(transfer): update test cases

fix(steps): fix error in deploy

refactor(steps): re-add change event

feat(utils): add change event constant to constants.ts

refactor(steps): correct requested changes

refactor(steps): replace indexOf with includes

build(deps-dev): bump vue-loader from 16.0.0-beta.5 to 16.0.0-beta.6

build(deps-dev): bump css-loader from 4.2.2 to 4.3.0

build(deps-dev): bump @swc-node/jest from 0.3.4 to 0.3.5

build(deps-dev): bump vue-loader from 16.0.0-beta.6 to 16.0.0-beta.7

test: hoist up mocks (#294)

* Add overlay component; Dialog component almost done

* feat(dialog): add use-lockscreen

* feat(dialog): coding completed awaiting tests

* feat(dialog): finish writing test cases

* fix test failures

* Address PR comments

* fallback some changes

fix(dialog): fix dialog exporting error (#298)

fix: fix lang type error (#299)

fix: fix eslint && typescript-eslint issue

fix(input): fix issues

https://github.com/ElemeFE/element/issues/18573

fix:Fit Avatar Container style

feat: inifinite scroll done

fix(infinite-scroll): wipe out sleep

fix: merge error

fix(infinite-scroll): should not trigger load when scroll up

Feat/carousel (#151)

* feat: init carousel component

* feat(carousel): init carousel component

* chore: add global dependencies

* chore: use prettier formate code

* feat(carousel): transfer logic - progress 40%

* feat(carousel): migrate logic - progress 50%

* feat(carousel): migrate logic - progress 55%

* feat(carousel): replenish doc

* feat(carousel): add utils

* feat(carousel): finish component logi

* feat(carousel): fix type error

* feat(carousel): test case 80%

* feat(carousel): migrate finish

* feat(carousel): more test cases

* feat(carousel): test case passed

* feat(carousel): fix CI dependencies collides

* feat(carousel): update yarn.lock

* feat(carousel): merge sub component

* feat(carousel): fix lose ctx attribute in buid env

* feat(carousel): finish spec

* feat(carousel): optimize code

* chore: update yarn lock

* feat(carousel): fall back lock file

* chore(carousel): fallback dep

* feat(carousel): update vue dep

* feat(carousel): update spec file

* feat(carousel): use async test

* feat(carousel): revert eslint modify

Co-authored-by: liik <385211478@qq.com>
Co-authored-by: liik <linyunqianpp@126.com>

fix(tag): fix click event of tag

feat(core): add jsx support

build: init build script

chore: update

1.0.1-alpha.0

chore: update

chore: update

chore: update

chore: update

1.0.1-alpha.1

chore: update

style(dropdown): improve dropdown style

partial fix

fix(popper): fix known bugs in popper

fix(dropdown): fix dropdown code based on popper's change

fix(popper): addressing comments

fix: Update install doc

chore: update

fix: Fix website code highlight

fix(color-picker): fix import tooltip path

test: the unit tests is done

test: fix tests of about 'show color picker'

fix: clear color then, drag mouse panel, but color do not change

Bumps [@storybook/html](https://github.com/storybookjs/storybook/tree/HEAD/app/html) from 6.0.20 to 6.0.21.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.0.21/app/html)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 3.10.1 to 4.0.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.0.0/packages/eslint-plugin)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump husky from 4.2.5 to 4.3.0

Bumps [husky](https://github.com/typicode/husky) from 4.2.5 to 4.3.0.
- [Release notes](https://github.com/typicode/husky/releases)
- [Changelog](https://github.com/typicode/husky/blob/master/CHANGELOG.md)
- [Commits](https://github.com/typicode/husky/compare/v4.2.5...v4.3.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump html-webpack-plugin from 4.3.0 to 4.4.1

Bumps [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin) from 4.3.0 to 4.4.1.
- [Release notes](https://github.com/jantimon/html-webpack-plugin/releases)
- [Changelog](https://github.com/jantimon/html-webpack-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jantimon/html-webpack-plugin/compare/v4.3.0...v4.4.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

ci: fix dependabot automerge

build(deps-dev): bump @vue/compiler-sfc from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps): bump vue from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps-dev): bump @swc-node/jest from 0.2.0 to 0.3.4

build(deps-dev): bump @storybook/addon-storysource from 6.0.20 to 6.0.21

build(deps-dev): bump @types/jest from 26.0.10 to 26.0.13

build(deps-dev): bump highlight.js from 10.1.2 to 10.2.0

build(deps-dev): bump sass-loader from 10.0.1 to 10.0.2

build(deps-dev): bump vue-router from 4.0.0-beta.7 to 4.0.0-beta.9

build(deps-dev): bump file-loader from 6.0.0 to 6.1.0

build(deps): bump dayjs from 1.8.34 to 1.8.35

build(deps-dev): bump lint-staged from 10.2.13 to 10.3.0

build(deps-dev): bump eslint from 7.7.0 to 7.8.1

build(deps-dev): bump @babel/core from 7.11.4 to 7.11.6

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.11.4 to 7.11.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.11.6/packages/babel-core)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

refactor(transfer): transfer component refactoring completed

docs(transfer): add docs files

test(transfer): add test file

refactor(transfer): correct requested changes

fix(transfer): rename d.ts to .ts

fix(transfer): remove unused declare

fix(transfer): update test cases

fix(steps): fix error in deploy

refactor(steps): re-add change event

feat(utils): add change event constant to constants.ts

refactor(steps): correct requested changes

refactor(steps): replace indexOf with includes

build(deps-dev): bump vue-loader from 16.0.0-beta.5 to 16.0.0-beta.6

build(deps-dev): bump css-loader from 4.2.2 to 4.3.0

build(deps-dev): bump @swc-node/jest from 0.3.4 to 0.3.5

build(deps-dev): bump vue-loader from 16.0.0-beta.6 to 16.0.0-beta.7

test: hoist up mocks (#294)

### feat: add dialog (#197)

* Add overlay component; Dialog component almost done

* feat(dialog): add use-lockscreen

* feat(dialog): coding completed awaiting tests

* feat(dialog): finish writing test cases

* fix test failures

* Address PR comments

* fallback some changes

fix(dialog): fix dialog exporting error (#298)

fix: fix lang type error (#299)

fix: fix eslint && typescript-eslint issue

fix(input): fix issues

https://github.com/ElemeFE/element/issues/18573

fix:Fit Avatar Container style

feat: inifinite scroll done

fix(infinite-scroll): wipe out sleep

fix: merge error

fix(infinite-scroll): should not trigger load when scroll up

Feat/carousel (#151)

* feat: init carousel component

* feat(carousel): init carousel component

* chore: add global dependencies

* chore: use prettier formate code

* feat(carousel): transfer logic - progress 40%

* feat(carousel): migrate logic - progress 50%

* feat(carousel): migrate logic - progress 55%

* feat(carousel): replenish doc

* feat(carousel): add utils

* feat(carousel): finish component logi

* feat(carousel): fix type error

* feat(carousel): test case 80%

* feat(carousel): migrate finish

* feat(carousel): more test cases

* feat(carousel): test case passed

* feat(carousel): fix CI dependencies collides

* feat(carousel): update yarn.lock

* feat(carousel): merge sub component

* feat(carousel): fix lose ctx attribute in buid env

* feat(carousel): finish spec

* feat(carousel): optimize code

* chore: update yarn lock

* feat(carousel): fall back lock file

* chore(carousel): fallback dep

* feat(carousel): update vue dep

* feat(carousel): update spec file

* feat(carousel): use async test

* feat(carousel): revert eslint modify

Co-authored-by: liik <385211478@qq.com>
Co-authored-by: liik <linyunqianpp@126.com>

fix(tag): fix click event of tag

feat(core): add jsx support

build: init build script

chore: update

1.0.1-alpha.0

chore: update

chore: update

chore: update

chore: update

1.0.1-alpha.1

chore: update

style(dropdown): improve dropdown style

partial fix

fix(popper): fix known bugs in popper

fix(dropdown): fix dropdown code based on popper's change

fix(popper): addressing comments

fix: Update install doc

chore: update

fix: Fix website code highlight

feat: Feature/timepicker && repeat-click directive (#289)

* feat: Feature/datepicker && repeat-click directive (#288)

* style: fix lint

* test: fix local test

* test: update test

* fix: update api to disabledHours

* chore: update

* chore: fix lint

feat: add calendar (#305)

* feat: add calendar

* chore: update doc

* chore: update

* chore: update

fix(locale): i18n support params

Fix/update website (#322)

* 1.0.1-alpha.2

* chore: update website logo

* chore: update website logo

chore: update readme (#325)

Feature/drawer (#248)

* feat(drawer): feature is done without test

* test(drawer): add unit test for drawer and fix drawer style

* fix(drawer): lint error for revert eslintrc

* feat(drawer): optimize appendtobody by use built-in teleport

* fix(drawer): resolve some review conservation

* fix(drawer): resolve nest drawer & use v-model for default visible prop

Co-authored-by: luxueyan <luxueyan@kaitongamc.com>

feat: bump vue version (#330)

* feat(core): bump vue version

* fix(steps): remove the usage of setupState

feat: init type

feat(hooks): add useAttrs hook (#317)

* feat(hooks): add useAttrs hook

* feat(hooks): undo binding class and style automatically

* test: remove unused import statement

ammend

feat(color-picker): add color-picker component

add color-picker component

perf: delete test.vue file, add type defintion

add 'IUseOptions' interface for provide and inject function.  del test.vue component

fix: add modelValue watch. fixed some problem for review code

feat: add unit test

fix: fixed unit test bug and components bug

fix(color-picker): add unit test and rebase from dev

Bumps [@storybook/html](https://github.com/storybookjs/storybook/tree/HEAD/app/html) from 6.0.20 to 6.0.21.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.0.21/app/html)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 3.10.1 to 4.0.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.0.0/packages/eslint-plugin)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump husky from 4.2.5 to 4.3.0

Bumps [husky](https://github.com/typicode/husky) from 4.2.5 to 4.3.0.
- [Release notes](https://github.com/typicode/husky/releases)
- [Changelog](https://github.com/typicode/husky/blob/master/CHANGELOG.md)
- [Commits](https://github.com/typicode/husky/compare/v4.2.5...v4.3.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump html-webpack-plugin from 4.3.0 to 4.4.1

Bumps [html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin) from 4.3.0 to 4.4.1.
- [Release notes](https://github.com/jantimon/html-webpack-plugin/releases)
- [Changelog](https://github.com/jantimon/html-webpack-plugin/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jantimon/html-webpack-plugin/compare/v4.3.0...v4.4.1)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

ci: fix dependabot automerge

build(deps-dev): bump @vue/compiler-sfc from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps): bump vue from 3.0.0-rc.9 to 3.0.0-rc.10

build(deps-dev): bump @swc-node/jest from 0.2.0 to 0.3.4

build(deps-dev): bump @storybook/addon-storysource from 6.0.20 to 6.0.21

build(deps-dev): bump @types/jest from 26.0.10 to 26.0.13

build(deps-dev): bump highlight.js from 10.1.2 to 10.2.0

build(deps-dev): bump sass-loader from 10.0.1 to 10.0.2

build(deps-dev): bump vue-router from 4.0.0-beta.7 to 4.0.0-beta.9

build(deps-dev): bump file-loader from 6.0.0 to 6.1.0

build(deps): bump dayjs from 1.8.34 to 1.8.35

build(deps-dev): bump lint-staged from 10.2.13 to 10.3.0

build(deps-dev): bump eslint from 7.7.0 to 7.8.1

build(deps-dev): bump @babel/core from 7.11.4 to 7.11.6

Bumps [@babel/core](https://github.com/babel/babel/tree/HEAD/packages/babel-core) from 7.11.4 to 7.11.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.11.6/packages/babel-core)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

refactor(transfer): transfer component refactoring completed

docs(transfer): add docs files

test(transfer): add test file

refactor(transfer): correct requested changes

fix(transfer): rename d.ts to .ts

fix(transfer): remove unused declare

fix(transfer): update test cases

fix(steps): fix error in deploy

refactor(steps): re-add change event

feat(utils): add change event constant to constants.ts

refactor(steps): correct requested changes

refactor(steps): replace indexOf with includes

build(deps-dev): bump vue-loader from 16.0.0-beta.5 to 16.0.0-beta.6

build(deps-dev): bump css-loader from 4.2.2 to 4.3.0

build(deps-dev): bump @swc-node/jest from 0.3.4 to 0.3.5

build(deps-dev): bump vue-loader from 16.0.0-beta.6 to 16.0.0-beta.7

test: hoist up mocks (#294)

* Add overlay component; Dialog component almost done

* feat(dialog): add use-lockscreen

* feat(dialog): coding completed awaiting tests

* feat(dialog): finish writing test cases

* fix test failures

* Address PR comments

* fallback some changes

fix(dialog): fix dialog exporting error (#298)

fix: fix lang type error (#299)

fix: fix eslint && typescript-eslint issue

fix(input): fix issues

https://github.com/ElemeFE/element/issues/18573

fix:Fit Avatar Container style

feat: inifinite scroll done

fix(infinite-scroll): wipe out sleep

fix: merge error

fix(infinite-scroll): should not trigger load when scroll up

Feat/carousel (#151)

* feat: init carousel component

* feat(carousel): init carousel component

* chore: add global dependencies

* chore: use prettier formate code

* feat(carousel): transfer logic - progress 40%

* feat(carousel): migrate logic - progress 50%

* feat(carousel): migrate logic - progress 55%

* feat(carousel): replenish doc

* feat(carousel): add utils

* feat(carousel): finish component logi

* feat(carousel): fix type error

* feat(carousel): test case 80%

* feat(carousel): migrate finish

* feat(carousel): more test cases

* feat(carousel): test case passed

* feat(carousel): fix CI dependencies collides

* feat(carousel): update yarn.lock

* feat(carousel): merge sub component

* feat(carousel): fix lose ctx attribute in buid env

* feat(carousel): finish spec

* feat(carousel): optimize code

* chore: update yarn lock

* feat(carousel): fall back lock file

* chore(carousel): fallback dep

* feat(carousel): update vue dep

* feat(carousel): update spec file

* feat(carousel): use async test

* feat(carousel): revert eslint modify

Co-authored-by: liik <385211478@qq.com>
Co-authored-by: liik <linyunqianpp@126.com>

fix(tag): fix click event of tag

feat(core): add jsx support

build: init build script

chore: update

1.0.1-alpha.0

chore: update

chore: update

chore: update

chore: update

1.0.1-alpha.1

chore: update

style(dropdown): improve dropdown style

partial fix

fix(popper): fix known bugs in popper

fix(dropdown): fix dropdown code based on popper's change

fix(popper): addressing comments

fix: Update install doc

chore: update

fix: Fix website code highlight

fix(color-picker): fix import tooltip path

test: the unit tests is done

test: fix tests of about 'show color picker'

fix: clear color then, drag mouse panel, but color do not change

perf: unit tests add _mount for short code

test: fixed unit tests bug

Bumps [@storybook/html](https://github.com/storybookjs/storybook/tree/HEAD/app/html) from 6.0.20 to 6.0.21.
- [Release notes](https://github.com/storybookjs/storybook/releases)
- [Changelog](https://github.com/storybookjs/storybook/blob/next/CHANGELOG.md)
- [Commits](https://github.com/storybookjs/storybook/commits/v6.0.21/app/html)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

build(deps-dev): bump @typescript-eslint/eslint-plugin

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 3.10.1 to 4.0.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v4.0.0/packages/eslint-plugin)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

fix: draggable.js change to typescript,and addEventListen change to on

Hades-li committed:
feat(color-picker): add color-picker component

add color-picker component

bastarder commented on packages/color-picker/src/components/alpha-slider.vue:
```diff
@@ -0,0 +1,144 @@
+
+    function getThumbTop() {
+      if (!props.vertical) return 0
+      const alpha = color.get('_alpha')
```
is here correctly ?
i saw the `color.ts` code
```
  get(prop: string) {
    return this['_' + prop]
  }
```
this line will become `color.__alpha` ?
emmm... where is `color` define?  maybe `props.color`?

Commit:
perf: delete test.vue file, add type defintion

add 'IUseOptions' interface for provide and inject function.  del test.vue component

bastarder commented on packages/color-picker/src/components/predefine.vue:
```diff
@@ -0,0 +1,71 @@
+        item.selected = color.compare(item)
+      })
+    })
+    watch(props.colors, newVal => {
```
can be reactively?
```suggestion
    watch(() => props.colors, newVal => {
```

Hades-li replied:
fixed:
```
watchEffect(() => {
      rgbaColors.value = parseColors(props.colors, props.color)
})
```

bastarder commented on packages/color-picker/src/components/predefine.vue:
```diff
@@ -0,0 +1,71 @@
+    watch(props.colors, newVal => {
+      rgbaColors.value = parseColors(newVal, props.color)
+    })
+    watch(props.color, newVal => {
```
ditto

Hades-li replied:
ditto

bastarder commented on packages/color-picker/src/components/sv-panel.vue:
```diff
@@ -0,0 +1,176 @@
+    // data
+    const cursorTop = ref<number>(0)
+    const cursorLeft = ref<number>(0)
+    const background = ref<string>('hsl(0, 100%, 50%)')
```
unnecessary type

Hades-li replied:
done

bastarder commented on packages/color-picker/src/components/sv-panel.vue:
```diff
@@ -0,0 +1,176 @@
+      colorValue,
+    }
+  },
+  /*data() {
```
remove comments

Hades-li replied:
done

bastarder commented on packages/color-picker/src/draggable.js:
```diff
@@ -0,0 +1,36 @@
+    }
+  }
+  const upFn = function(event) {
+    document.removeEventListener('mousemove', moveFn)
```
use `on,off`

Hades-li replied:
wait...,i will change draggable.js to ts and resolve this code

Hades-li replied:
done

bastarder commented on packages/color-picker/src/index.vue:
```diff
@@ -0,0 +1,256 @@
+  <el-tooltip
+    v-model="showPicker"
+    effect="light"
+    :manual="true"
```
```suggestion
    manual
```

Hades-li replied:
done

bastarder commented on packages/color-picker/src/index.vue:
```diff
@@ -0,0 +1,256 @@
+    <template #content>
+      <div
+        v-click-outside="hide"
+        style="width: 300px"
```
?

Hades-li replied:
done

bastarder commented on packages/color-picker/src/index.vue:
```diff
@@ -0,0 +1,256 @@
+            ref="hue"
+            :color="color"
+            vertical
+            style="float: right;"
```
?

Hades-li replied:
ditto

bastarder commented on packages/color-picker/src/index.vue:
```diff
@@ -0,0 +1,256 @@
+    ClickOutside,
+  },
+  props: {
+    modelValue: String,
```
modelValue will not be reactively any where?
i saw the element-ui code, they watch value's change

Hades-li replied:
done

bastarder commented on packages/color-picker/src/test.vue:
```diff
@@ -0,0 +1,16 @@
+<template>
```
? remove useless component file

Hades-li replied:
deleted

bastarder reviewed (commented):
you should add unit test

Hades-li committed:
fix: add modelValue watch. fixed some problem for review code

Hades-li commented on packages/color-picker/src/components/alpha-slider.vue:
```diff
@@ -0,0 +1,144 @@
+      ref="bar"
+      class="el-color-alpha-slider__bar"
+      :style="{
+        background: background
```
done

Hades-li commented:
this pr is closed。change to new pr #375
